R/database.R

```diff
@@ -2628,3 +2630,385 @@ get_gene_expression = function(metadata,
+#'
+#' my_bedpe_df = get_manta_sv_by_sample(these_samples_metadata = my_metadata,
+#'                                      min_vaf = 0.05,
+#'                                      min_score = 30,
+#'                                      pass = FALSE,
+#'                                      with_chr_prefix = TRUE)
+#'
+get_manta_sv_by_sample = function(min_vaf = 0.1,
+                                  min_score = 40,
+                                  pass = TRUE,
+                                  this_sample_id,
+                                  these_samples_metadata,
+                                  with_chr_prefix = FALSE,
+                                  from_flatfile = TRUE,
+                                  projection = "grch37",
+                                  all_sv_grch37,
+                                  all_sv_hg38){
+
+  #check remote configuration
+  remote_session = check_remote_configuration(auto_connect = TRUE)
+
+  #retrieve sample ID from metadata, or get metadata for provided sample ID
+  #get sample ID from metadata and check if metadata contains more than 1 row
+  if(missing(this_sample_id)){
+    if(!missing(these_samples_metadata)){
+      this_sample_id = pull(these_samples_metadata, sample_id)
+      if(!nrow(these_samples_metadata==1)){
+        stop("Your metadata seems to be having more than one sample (or zero samples), please refer to get_manta_sv_by_samples if you indeed are interested in returning results for multiple samples...")
+      }
+    }
+  }
+
+  #get metadata and subset based on the provided sample ID
+  if(!missing(this_sample_id)){
+    if(missing(these_samples_metadata)){
+      these_samples_metadata = get_gambl_metadata() %>%
+        dplyr::filter(sample_id == this_sample_id)
+      if(!nrow(these_samples_metadata==1)){
+        stop("Fetched metadata contains more than one row, or the data frame is empty (subset to the provided sample ID). Considder giving this funciton an already subset metadata with the sample ID of interest...")
+      }
+    }
+  }
+
+  #last check
+  if(missing(this_sample_id) && missing(these_samples_metadata)){
+    stop("please provide either a sample ID (this_sample_id) or an already subset metadata (these_samples_metadata)...")
+  }
+
+  #get wildcards
+  tumour_sample_id = this_sample_id
+  unix_group = pull(these_samples_metadata, unix_group)
+  seq_type = pull(these_samples_metadata, seq_type)
+  genome_build = projection
+  pairing_status = pull(these_samples_metadata, pairing_status)
+
+  if(pairing_status == "matched"){
+    normal_sample_id = pull(these_samples_metadata, normal_sample_id)
+  }else{
+    normal_sample_id = config::get("unmatched_normal_ids")[[unix_group]][[seq_type]][[genome_build]]
+  }
+
+  if(from_flatfile){
+    if(!remote_session){
+      path_template = config::get("results_flatfiles")$sv_manta$template
+      path_template = paste0(config::get("project_base"), path_template)
+      bedpe_path = glue::glue(path_template)
+
```

Comment:
The logic here seems to be backwards. This will use the individual flatfiles as long as they exist even though most of those results are already in the merge. It should be the other way around: default to using what's in the merge and resort to the flatfiles if the sample is not in the merge.

Reply:
Ok, I am with you. A misunderstanding on my part. This has been addressed.